common/lib/client/realtimechannel.js

```diff
@@ -423,22 +437,37 @@ var RealtimeChannel = (function() {
 
 		case actions.MESSAGE:
 			var messages = message.messages,
+				firstMessage = messages[0],
+				lastMessage = messages[messages.length - 1],
 				id = message.id,
 				connectionId = message.connectionId,
 				timestamp = message.timestamp;
 
+			if(firstMessage.extras && firstMessage.extras.delta && firstMessage.extras.delta.from !== this.lastPayloadMessageId) {
+				Logger.logAction(Logger.LOG_MAJOR, 'RealtimeChannel.onMessage()', 'Delta message decode failure - previous message not available.');
+				this.startDecodeFailureRecovery();
+				break;
+			}
+
 			for(var i = 0; i < messages.length; i++) {
 				var msg = messages[i];
 				try {
 					Message.decode(msg, this.encodingDecodingContext);
 				} catch (e) {
 					/* decrypt failed .. the most likely cause is that we have the wrong key */
 					Logger.logAction(Logger.LOG_MINOR, 'RealtimeChannel.onMessage()', e.toString());
+					if(e.code === 40018) {
```

Comment:
@paddybyers, should I add this to https://github.com/ably/ably-common/blob/master/protocol/errors.json?

Reply:
Yes please 👍

Reply:
https://github.com/ably/ably-common/pull/48